java-checks/src/main/java/org/sonar/java/checks/IdenticalOperandOnBinaryExpressionCheck.java

```diff
@@ -69,18 +79,46 @@
       .add(Tree.Kind.OR)
       .add(Tree.Kind.CONDITIONAL_AND)
       .add(Tree.Kind.CONDITIONAL_OR)
+      .add(Tree.Kind.METHOD_INVOCATION)
       .build();
   }
 
   @Override
   public void visitNode(Tree tree) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776)